src/utils/selectors.ts

```diff
@@ -124,6 +126,23 @@ export const isUserPurchaseHistoryRequestedSelector: SparkleSelector<boolean> =
   "userPurchaseHistory"
 );
 
+const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+export const unreadMessagesSelector = (state: RootState) => {
+  const user = authSelector(state);
+  const privateChats = privateChatsSelector(state) ?? [];
+
+  return privateChats.some(
+    (message) =>
+      message.from !== user?.uid &&
+      message.deleted !== true &&
+      message.type === "private" &&
+      message.ts_utc.seconds > HIDE_BEFORE &&
+      message.isRead === false
+  );
+};
```

Comment:
Here's an example of a 'smarter' selector. Instead of selecting all of the data, then filtering it once we get to the component, by doing it in the selector we will get extra memoisation/etc (namely that our component will only need to re-render if the value emitted by the selector changes, not the 'dependencies' we use to calculate that final value.

Reply:
[OOS/future] Yes, and as we pull non-domain logic out of the code, it will be great to change from doing `useSelector(somethingSelector)` to `useAThingWeNeed` so the selector and `useFirestoreConnect` can go into one place; to make it easier to change how we get data in future.

Reply:
Definitely! And I believe we could also put the selector first, then use the 'does this exist' helpers or similar to decide if we even need to 'connect' the query at that point, so that it automagically knows if it needs to, and that is abstracted away from us so we just need to call `useTheThing` and not have to 🧠  about it beyond that.